app/models/application_form.rb

```diff
@@ -173,6 +180,7 @@ def equality_and_diversity_answers_provided?
   end
 
   after_update :geocode_address_and_update_region_if_required
+  after_save :update_adviser_availability, if: :saved_change_to_adviser_status?
```

Comment:
currently reviewing with @tomas-stefano. 

could this and the `update_adviser_availability` method be moved to a service class? `ApplicationForm` is probably the most core class in the application, and it'd be better if it didn't have a detailed knowledge of `Adviser:: SignUpAvailability`. if this were in a service class, we could also drop the `after_save`.

makes sense to keep the `enum` here.

Reply:
plus since `SignUpAvailability` updates the `ApplicationForm`, we have a cyclic dependency.